tests/test_cl2hpp.cpp

```diff
@@ -2516,50 +2560,131 @@ static cl_int clGetDeviceInfo_extended_versioning(
         }
         return CL_SUCCESS;
     }
-    case CL_DEVICE_EXTENSIONS_WITH_VERSION_KHR:
+    case CL_DEVICE_EXTENSIONS_WITH_VERSION:
     {
-        static cl_name_version_khr extension = {
-            CL_MAKE_VERSION_KHR(10, 11, 12),
+        static cl_name_version extension = {
+            CL_MAKE_VERSION(10, 11, 12),
             "cl_dummy_extension",
         };
-        if (param_value_size == sizeof(cl_name_version_khr) && param_value) {
-            *static_cast<cl_name_version_khr*>(param_value) = extension;
+        if (param_value_size == sizeof(cl_name_version) && param_value) {
+            *static_cast<cl_name_version*>(param_value) = extension;
         }
         if (param_value_size_ret) {
             *param_value_size_ret = sizeof(extension);
         }
         return CL_SUCCESS;
     }
-    case CL_DEVICE_ILS_WITH_VERSION_KHR:
+    case CL_DEVICE_ILS_WITH_VERSION:
     {
-        static cl_name_version_khr il = {
-            CL_MAKE_VERSION_KHR(20, 21, 22),
+        static cl_name_version il = {
+            CL_MAKE_VERSION(20, 21, 22),
             "DUMMY_IR",
         };
-        if (param_value_size == sizeof(cl_name_version_khr) && param_value) {
-            *static_cast<cl_name_version_khr*>(param_value) = il;
+        if (param_value_size == sizeof(cl_name_version) && param_value) {
+            *static_cast<cl_name_version*>(param_value) = il;
         }
         if (param_value_size_ret) {
             *param_value_size_ret = sizeof(il);
         }
         return CL_SUCCESS;
     }
-    case CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION_KHR:
+    case CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION:
     {
         // Test no built-in kernels:
         if (param_value_size_ret) {
             *param_value_size_ret = 0;
         }
         return CL_SUCCESS;
     }
+    case CL_DEVICE_OPENCL_C_ALL_VERSIONS:
+    {
+        static cl_name_version opencl_c = {
+            CL_MAKE_VERSION(30, 31, 32),
+            "OpenCL C",
+        };
+        if (param_value_size == sizeof(cl_name_version) && param_value) {
+            *static_cast<cl_name_version*>(param_value) = opencl_c;
+        }
+        if (param_value_size_ret) {
+            *param_value_size_ret = sizeof(opencl_c);
+        }
+        return CL_SUCCESS;
+    }
+    case CL_DEVICE_OPENCL_C_FEATURES:
+    {
+        static cl_name_version opencl_c_features[] = {
+            {
+                CL_MAKE_VERSION(40, 41, 42),
+                "__opencl_c_feature",
+            },
+            {
+                CL_MAKE_VERSION(40, 43, 44),
+                "__opencl_c_fancy_feature",
+            },
+        };
+        if (param_value_size == sizeof(opencl_c_features) && param_value) {
+            cl_name_version* feature = static_cast<cl_name_version*>(param_value);
+            const int numFeatures = sizeof(opencl_c_features) / sizeof(opencl_c_features[0]);
```

Comment:
Do we not have an `ARRAY_SIZE` macro already? I couldn't find one, maybe it's a good excuse to introduce one.

Reply:
Added.